modules/crm_tools/src/Element/RoleIcon.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Drupal\crm_tools\Element;
+
+use Drupal\Component\Utility\Html as HtmlUtility;
+use Drupal\Core\Render\Element\RenderElement;
+use Drupal\Core\Render\Markup;
+use Drupal\Core\Template\Attribute;
+
+/**
+ * Provides a render element for any HTML tag, with properties and value.
+ *
+ * Properties:
+ * - #icon: The icon name to render.
+ * - #attributes: (array, optional) HTML attributes to apply to the icon. The
+ *   attributes are escaped, see \Drupal\Core\Template\Attribute.
+ * - #color: (string, optional) A string for the background hex color.
```

Comment:
This doesn't need to be a hex, for example an `rgba()` would also be fine? Just needs to be a CSS compatible colour definition?

Reply:
Also, do we want to allow setting the `bg_color` vs `fill`/`color`? Even if our roles stuff only uses bg?